### PR TITLE
[yaml] Enhance YAML API docs

### PR DESCRIPTION
**Pull request by Polber**
The current YAML API refernce docs look like the following:
image

This PR adds formatting to make the docs look more like the Sphinx-themed PyDocs used by the Python SDK:
image

This change ~~not only~~ adds some color and easier-to-read formatting ~~, but also separates the transforms into individual pages to reduce clutter~~. This should hopefully make the user-experience a bit better.

UPDATE: the doc will remain as one page, but will include the new side panel and formatting changes.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Polber** [comment]
R: @robertwb

**Review by mravi — APPROVED** [review]

**Comment by liferoad** [comment]
Can you check https://github.com/apache/beam/actions/runs/11579341692/job/32235315129?pr=32825?

File: sdks/python/apache_beam/yaml/tests/map.yaml [hunk 1]
```diff
@@ -31,8 +31,8 @@ pipelines:
             append: true
             fields:
               # TODO(https://github.com/apache/beam/issues/32832): Figure out why Java sometimes re-orders these fields.
-              literal_int: 10
               named_field: element
+              literal_int: 10
```
Comment (robertwb) [inline sdks/python/apache_beam/yaml/tests/map.yaml:35]: We should try to get to the bottom of this... File a bug?

**Review by robertwb — APPROVED** [review]
This looks lots nicer.

File: sdks/python/apache_beam/yaml/tests/map.yaml (hunk 1 above)
Comment (damccorm) [reply sdks/python/apache_beam/yaml/tests/map.yaml:35]: Agreed, this isn't super sustainable. I think we can just use https://github.com/apache/beam/issues/32832

**Review by damccorm — APPROVED** [review]
LGTM, thanks!

**Comment by damccorm** [comment]
Test failures are pretty clearly unrelated, so I'm going to merge

**Pull request merged**